app/models/task/anticipated_user_notification_list.rb

```diff
@@ -30,7 +30,7 @@ def generate
       suppliers.find_each do |supplier|
         next if supplier.active_frameworks.empty?
 
-        supplier.users.each do |user|
+        supplier.active_users.each do |user|
```

Comment:
I think you can use `find_each` here as well.

Reply:
The `active_users` association is preloaded as part of the query in the `suppliers` method so batch loading them wouldn't make much sense. Besides which the users-per-suppliers count is always going to be relatively small (unlikely to go beyond double figures), so batching them would have little effect (default batch size is a 1000 I think?)